### DISCUSSION TITLE
Change CWD

### DISCUSSION THREAD

**Discussion by sudocpMATHdotPY** [pyinstaller/pyinstaller#7056] [Help]
I am writing a simple program that runs other executable files, but I need it to run those files inside a certain directory. 
I have tried `os.chdir` and `subprocess.Popen(app, cwd = folder)` it works in the IDE, but fails when running the executable (I am using absolute paths).

How do I reliably change the working directory?

**Comment by rokm** [comment]
If you are trying to change the directory using relative paths, then you are implicitly assuming that you know where your *current* current-working-directory is. In a frozen application, you don't really know that, because executable could be launched from an arbitrary location. So you need to anchor your relative paths to either `__file__` or to `sys.executable` to obtain an absolute path.

Once you have an absolute path, changing directory should work as expected.
